### ISSUE TITLE
Pattern for x64 architecture matching wrong function

### ISSUE THREAD

**Issue by RafaelDiasCampos** [NVISOsecurity/disable-flutter-tls-verification#34]
When trying to intercept requests in an x86_64 Android Studio emulator, the script found a pattern match, but patching the function broke TLS validation even when not using a proxy. Analyzing the libflutter library, it seems that the last pattern for x64 (55 41 57 41 56 41 55 41 54 53 48 83 EC 38 C6 02 50 48 8B AF A? 00 00 00 48 85 ED 74 7? 48 83 7D 00 00 74) is matching the function [ssl_crypto_x509_session_verify_cert_chain](https://github.com/google/boringssl/blob/58f3bc83230d2958bb9710bc910972c4f5d382dc/ssl/ssl_x509.cc#L324) instead of the desired [ssl_verify_peer_cert](https://github.com/google/boringssl/blob/master/ssl/handshake.cc#L323), as can be seen from the following screenshots:

Decompilation of function found at pattern:
![image](https://github.com/user-attachments/assets/681566b7-8525-4abd-8b96-55208fc1506c)
![image](https://github.com/user-attachments/assets/d64121f1-56b0-4528-88c5-d414a349b551)

String literals in the decompiled function:
![image](https://github.com/user-attachments/assets/cc2e87f3-ed47-4086-9b09-91c044adcde5)
![image](https://github.com/user-attachments/assets/3325b2cd-10cb-4428-949b-2398008cf36b)

The ssl_verify_peer_cert was found at address 0x6a80c6 (007a80c6 in Ghidra), with the pattern "55 41 57 41 56 41 55 41 54 53 50 49 89 fe 48 8b 1f 48 8b 43 30 4c 8b b8 d0 01 00 00 4d 85 ff 74 12 4d 8b a7 90 00 00 00 4d 85 e4 74 4a 49 8b 04 24 eb 46". In my opinion, it would be best to improve the existing pattern to remove incorrectly identifying and patching ssl_crypto_x509_session_verify_cert_chain, but it's not necessary, as even with both functions being modified TLS verification gets disabled:

Both functions were modified, but request interception still works
![image](https://github.com/user-attachments/assets/fa9ce8c7-35b6-4e71-a6c0-c5f8d634cef9)

As a last note, during my tests, I realized that replacing ssl_crypto_x509_session_verify_cert_chain to return 1 (true) can also be used to disable TLS verification, without changing ssl_verify_peer_cert.

TLDR: New pattern: "55 41 57 41 56 41 55 41 54 53 50 49 89 fe 48 8b 1f 48 8b 43 30 4c 8b b8 d0 01 00 00 4d 85 ff 74 12 4d 8b a7 90 00 00 00 4d 85 e4 74 4a 49 8b 04 24 eb 46".

**Comment by RafaelDiasCampos** [comment]
Including libflutter.so library
[libflutter.zip](https://github.com/user-attachments/files/17016886/libflutter.zip)
